src/components/task/uc/deleteUserData.uc.unit.test.js

```diff
@@ -0,0 +1,332 @@
+				removeGroupSubmissionsConnectionsForUserStub.restore();
+				// then
+				expect(result.complete).to.be.true;
+				expect(result.trashBinData).to.be.an('object');
+				expect(result.trashBinData).to.haveOwnProperty('scope');
+				expect(result.trashBinData).to.haveOwnProperty('data');
+				expect(result.trashBinData.scope).to.be.equal('submissions-shared');
+				expect(result.trashBinData.data.map(idToString)).to.have.members(
+					sharedSubmissions.map((hw) => idToString(hw._id))
+				);
+			});
+
+			// user not exist
+			it('should return empty data if there is no shared submission', async () => {
+				// given
+				const otherUserId = new ObjectId();
+				const { findGroupSubmissionIdsByUserStub, removeGroupSubmissionsConnectionsForUserStub } = initStubs({
+					teamMember: userId,
+				});
+				// when
+				const { removeConnectionToSharedSubmissions } = mapByName(deleteUserRelatedData());
+				const result = await removeConnectionToSharedSubmissions(otherUserId);
+				findGroupSubmissionIdsByUserStub.restore();
+				removeGroupSubmissionsConnectionsForUserStub.restore();
+				// then
+				expect(result.complete).to.be.true;
+				expect(result.trashBinData.data).to.be.an('array').that.is.empty;
+				expect(result.trashBinData.scope).to.be.equal('submissions-shared');
+			});
+		});
+
+		describe('deletePrivateUserHomeworks', () => {
+			const initStubs = ({ teacherId }) => {
+				const deletePrivateHomeworksFromUserStub = sinon.stub(repo, 'deletePrivateHomeworksFromUser');
+				const privateHomeworks = [
+					{ _id: new ObjectId(), teacherId, private: true },
+					{ _id: new ObjectId(), teacherId, private: true },
+				];
+				deletePrivateHomeworksFromUserStub.withArgs(userId).returns(getExpectedDeleteMany(2));
+
+				const findPrivateHomeworksByUserStub = sinon.stub(repo, 'findPrivateHomeworksByUser');
+				findPrivateHomeworksByUserStub.callsFake(() => []);
+				findPrivateHomeworksByUserStub.withArgs(userId).returns(privateHomeworks);
+
+				return { findPrivateHomeworksByUserStub, deletePrivateHomeworksFromUserStub, privateHomeworks };
+			};
+
+			// true work
+			it('should return deleted private homeworks', async () => {
+				// given
+				const { findPrivateHomeworksByUserStub, deletePrivateHomeworksFromUserStub, privateHomeworks } = initStubs({
+					teacherId: userId,
+				});
+				// when
+				const { deletePrivateUserHomeworks } = mapByName(deleteUserRelatedData());
+				const result = await deletePrivateUserHomeworks(userId);
+				findPrivateHomeworksByUserStub.restore();
+				deletePrivateHomeworksFromUserStub.restore();
+				// then
+				expect(result.complete).to.be.true;
+				expect(result.trashBinData).to.be.an('object');
+				expect(result.trashBinData).to.haveOwnProperty('scope');
+				expect(result.trashBinData).to.haveOwnProperty('data');
+				expect(result.trashBinData.scope).to.be.equal('homeworks-private');
+				expect(result.trashBinData.data).to.have.members(privateHomeworks);
```

Comment:
thi is repeating code, a expectValidTrashBinResult(result, scopeName, dataToHaveMembers) could be used instead. maybe additional an emptyTrashbinresult check too, to ensure they match each other